MobiusCore/Source/EffectHandlers/EffectHandler.swift

```diff
@@ -17,16 +17,44 @@
 // specific language governing permissions and limitations
 // under the License.
 
-// Primitive EffectHandler
-public struct EffectHandler<Effect, Event> {
-    public let handle: (Effect, @escaping Consumer<Event>) -> Void
-    public let disposable: Disposable
+public protocol EffectHandler {
+    associatedtype Effect
+    associatedtype Event
+
+    /// Handle an `Effect`.
+    /// To output events, call `response.send`
+    /// When you are done handling `input`, be sure to call `response.end()` to prevent memory leaks.
+    ///
+    /// Return a `Disposable` which tears down any resources that is being used by this effect handler.
```

Comment:
This should clearly document which orderings of `dispose` and `end` are allowed.